### PR TITLE
[DEBUG] Add test for C++ exception error

### PR DESCRIPTION
**Pull request by xylar**
This is a test PR to see whether the test that @gillins added in #233 passes for 1.14.4

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@gillins, osx-64 is passing here, too.  So I don't think we have an explanation for why kealib is failing in https://github.com/conda-forge/kealib-feedstock/pull/65

**Comment by gillins** [comment]
Sigh. Thanks @xylar. @petebunting any chance you'll have some time to look into this some more?

File: recipe/build.sh [hunk 1]
```diff
@@ -158,3 +158,9 @@ if [[ ("$target_platform" != "linux-ppc64le") && \
   # https://forum.hdfgroup.org/t/hdf5-1-10-long-double-conversions-tests-failed-in-ppc64le/4077
   make check RUNPARALLEL="mpiexec -n 2"
 fi
+
+# test for hdf5 C++ exceptions
+if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" != "1" || "${CROSSCOMPILING_EMULATOR}" != "" ]]; then
+    $CXX ${CXXFLAGS} $RECIPE_DIR/testhdf5exc.cpp -o testhdf5exc -L$PREFIX/lib -lhdf5_cpp -lhdf5 -I$PREFIX/include
```
Comment (hmaarrfk) [inline recipe/build.sh:164]:
```suggestion
    ${CXX} ${CXXFLAGS} ${LDFLAGS} $RECIPE_DIR/testhdf5exc.cpp -o testhdf5exc -lhdf5_cpp -lhdf5 -I$PREFIX/include
```

You can also follow the style and add the test to the test section:
https://github.com/xylar/hdf5-feedstock/blob/add-exception-catch-test/recipe/run_test.sh#L23

Comment (xylar) [reply recipe/build.sh]: Agreed, I will move it if we decide we want to keep the test.

**Comment by gillins** [comment]
Update: `valgrind` shows no errors running the `kealib` test on Linux x86_64 with `hdf5-1.14`, so nothing useful there....

**Comment by xylar** [comment]
@gillins, no surprise.  The issue seems to be specific to clang.  My understanding based on some googling is that it's related to mixing up the system vs. the conda-forge (or in my google searches the homebrew) version of a core c++ library (can't recall which).  Anyway, seems both osx and clang specific.

**Comment by gillins** [comment]
Ah ok, so might be a problem for other packages too? Unfortunately I no longer have a Mac to run other tests...

**Comment by xylar** [comment]
I'm going to close this as I don't think it was helpful.

**Pull request closed without merging**